lib/entry/crossDCsStreamPersistence.js

```diff
@@ -0,0 +1,275 @@
+                if (newEntry) {
+                    logger.debug(`====> [streamPersistenceCheck] [${entryId}] STOPPED STREAMING expiration time set to [${nowStr}]`);
+                    playlist = playlist.replace(this._expireTimeRegex, `$1${now}`);
+                }
+                if (entryIsLive) {
+                    logger.debug(`====> [streamPersistenceCheck] [${entryId}] is LIVE in other DC, presentation end time set to [${presentationTimeStr}]`);
+                    playlist = playlist.replace(this._presentationEndTimeRegex, `$1${presentationEndTime}`);
+                } else {
+                    ended = true;
+                    logger.debug(`====> [streamPersistenceCheck] [${entryId}] ENDED. Presentation end time set to [${nowStr}]`);
+                    playlist = playlist.replace(this._presentationEndTimeRegex, `$1 ${now}`);
+                }
+                return playlist;
+            })
+            .then((playlist) => {
+                return qio.write(this._playlistsData[entryId].path, playlist);
+            })
+            .then(() => {
+                if (ended) {
+                    return this.removeEntry(entryId, true);
+                } else {
+                    return Q.resolve(false);
+                }
+            })
+            .catch((err) => {
+                logger.error(`persistence check failed for [${entryId}]. error: [${ErrorUtils.error2string(err)}]`);
+                return Q.reject(err);
+            });
+    }
+
+    addEntry(entryId) {
+
+        logger.debug(`====> [addEntry] adding [${entryId}] to persistence list`);
+
+        if (this._entriesToProcess[entryId]) {
+            logger.debug("====> [addEntry][%s] entry is already in stream persistence list, ignoring request", entryId);
+            return Q.resolve();
+        } else {
+            this._entriesToProcess[entryId] = entryId;
+            this._playlistsData[entryId] = {
+                path: path.join(persistenceFormat.getEntryBasePath(entryId), persistenceFormat.getMasterManifestName()),
+                lastProcessedTime: Date.now()
+            };
+            return this.streamPersistenceCheck(entryId, true)
+                .then((ended) => {
+                    logger.info(`====> [addEntry][${entryId}] successfully added to stream persistence list (ended=${ended}), [playlistPath=${this._playlistsData[entryId]}]`);
+
+                    /*         // todo: rethink flash operation location and timing...
+                     return this.flushEntryList(this._entriesToProcess)
+                     .catch((err)=> {
+                     logger.error(`error restoring stopped entries list, error: [${ErrorUtils.error2string(err)}]`);
+                     return Q.resolve(true);
+                     });*/
+                })
+                .catch((err) => {
+                    logger.info(`====> [init] error restoring stopped entries list, error: [${ErrorUtils.error2string(err)}]`);
```

Comment:
Init purpose is to restore all stopped entries from file when LiveController starts. Never the less I added log with entry id where relevant.